src/client/components/common/StyledToast.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { FC } from 'react'
+import {
+  Box,
+  Flex,
+  HStack,
+  Icon,
+  IconButton,
+  RenderProps,
+  Spacer,
+  Text,
+  useMultiStyleConfig,
+  useToast,
+} from '@chakra-ui/react'
+import { BiCheckCircle, BiErrorCircle, BiX } from 'react-icons/bi'
+
+interface StyledToastProps extends RenderProps {
+  message: string
+  status: 'error' | 'success' | 'warning'
+}
+
+export const StyledToast: FC<StyledToastProps> = ({
+  message,
+  status,
+  onClose,
+}: StyledToastProps) => {
+  const styles = useMultiStyleConfig('StyledToast', { colorScheme: status })
+
+  const getIcon = (status: 'error' | 'success' | 'warning') => {
+    if (status === 'error' || status === 'warning') return BiErrorCircle
+    if (status === 'success') return BiCheckCircle
+    return BiCheckCircle
+  }
+
+  return (
+    <Flex
+      sx={styles.boxContainer}
+      role="alert"
+      aria-live="assertive"
+      aria-atomic="true"
+    >
+      <Box sx={styles.toastBox}>
+        <HStack spacing={2} sx={styles.container}>
+          <Icon
+            as={getIcon(status)}
+            sx={styles.icon}
+            title={`${status}-icon`}
+          />
+          <Text sx={styles.message}>{message}</Text>
+          <Spacer />
+          <IconButton
+            icon={<BiX />}
+            sx={styles.closeButton}
+            aria-label="close"
+            aria-hidden
+            variant="link"
+            onClick={onClose}
+          />
+        </HStack>
+      </Box>
+    </Flex>
+  )
+}
+
+export const useStyledToast = () => {
+  const toast = useToast({
+    position: 'top',
+    duration: 6000,
```

Comment:
Perhaps we can have the `render` function as part of this and just keep the same interface as the original `useToast`? `useStyledToast` should take in `UseToastOptions`. This way, the users of a styled toast do not need to be aware of the actual component they are using.

Reply:
Ordinarily, I would have used the `render` function to hide the custom rendered component, as suggested above. However, I had to resort to this particular method as the render props provided through the render function do not include the title, description or status data passed into the toast prop, which means that we have to specify our own props through the custom component.

Reply:
Note that the use of a custom toast component also excludes us from using a fair bit of the Chakra Toast API, so this is something we might have to bear with if we want to implement our own custom toast modals.

Reply:
How about something like this:

```ts
export const useStyledToast = (
  options?: UseToastOptions
): ReturnType<typeof useToast> => {
  const toast = useToast({
    position: 'top',
    duration: 6000,
    ...options,
  })

  const toastImpl = (opts?: UseToastOptions) => {
    const status = opts?.status ?? 'info'
    const description = opts?.description ?? ''

    return toast({
      render: (props) => (
        <StyledToast status={status} message={description} {...props} />
      ),
    })
  }

  toastImpl.close = toast.close
  toastImpl.closeAll = toast.closeAll
  toastImpl.isActive = toast.isActive
  toastImpl.update = toast.update

  return toastImpl
}
```